sdk/src/main/java/com/google/cloud/dataflow/sdk/io/BoundedSource.java

```diff
@@ -133,6 +133,95 @@ public Double getFractionConsumed() {
+     * consumed parallelism is 0.
+     * <li>For a non-empty, finished reader (in which the call to {@link #start} returned true and
+     * a call to {@link #advance} has returned false), the value returned is at must be at least 1
+     * and should equal the total parallelism in the source. For the two examples above, the
+     * 50-record file should have a consumed parallelism of 50.
+     * </ul>
+     *
+     * <p>A reader that is implemented using a {@link org.apache.beam.sdk.io.range.RangeTracker} is
+     * encouraged to delegate this function to its {@code getParallelismConsumed} method, if
+     * implemented.
+     *
+     * <p>Defaults to {@link #PARALLELISM_UNKNOWN}.
+     *
+     * @see #getParallelismRemaining()
+     */
+    public long getParallelismConsumed() {
+      return PARALLELISM_UNKNOWN;
+    }
+
+    /**
+     * Returns the total amount of parallelism in the unread range of this reader's current
+     * {@link BoundedSource} (as would be returned by {@link #getCurrentSource}). This function
+     * should be implemented only <strong>in addition to {@link #getParallelismConsumed()}</strong>
+     * and only if <em>an exact value can be returned</em>.
+     *
+     * <p>The "parallelism" remaining does not necessarily equal the number of <em>records</em>,
+     * but rather the number of distinct (ideally non-empty) <em>sources</em> that could be
+     * produced by calls to {@link #splitAtFraction}. For a {@link BoundedReader BoundedReader}
+     * implemented using a {@link org.apache.beam.sdk.io.range.RangeTracker}, this would correspond
+     * to the number of split points remaining.
+     *
+     * <p>Consider the following two examples. An input that can be read in parallel down to the
+     * individual records, such as {@link CountingSource#upTo}, is called "perfectly splittable".
+     * Conversely, an example of a non-perfectly splittable input is a block-compressed file format
+     * such as {@link AvroIO}, in which a block of records has to be read as a whole, but different
+     * blocks can be read in parallel.
+     *
+     * <ul>
+     * <li>Any {@link BoundedReader reader} should return a value of at least 1 indicating the
+     * source it is currently reading.
+     * <li>When processing record #30 (starting at 1) out of 50 in a perfectly splittable 50-record
+     * input, this value should be 21 (20 remaining + 1 current).
+     * <li>If we are reading through block 3 in a block-compressed file consisting of 5 blocks,
+     * this value should be 3 (since blocks 4 and 5 can be processed in parallel by new readers
+     * produced via dynamic work rebalancing, while the current reader remains processing block 3).
+     * <li>If we are reading through the last block in a block-compressed file, or reading or
+     * processing the last record in a perfectly splittable input, this value should be 1: apart
+     * from the current task, no additional remainder can be split off.
```

Comment:
Add that if the source is not splittable in general, the returned value should be 1.